### PR TITLE
feat: add clear button to search input

### PR DESCRIPTION
**Pull request by pregnantboy**
## Changes
- add clear button in search input
- refactor executions search input with filter into separate component since it's getting too big

### CODE REVIEW

**Comment by pregnantboy** [comment]
* **#774** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/774?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#758** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/758?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#773](https://github.com/opengovsg/plumber/pull/773) <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/773?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#757** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/757?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#756** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/756?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pregnantboy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/src/components/DebouncedSearchInput/index.tsx [hunk 1]
```diff
@@ -13,31 +18,54 @@ export default function DebouncedSearchInput({
...
+    setTempSearchValue('')
+    onChange('')
+    handleSearchInputChangeDebounced.cancel()
+  }, [onChange, handleSearchInputChangeDebounced])
+
   return (
     <InputGroup>
       <InputLeftElement h="100%">
-        <Icon as={BiSearch} style={{ height: 20, width: 20 }} />
+        <Icon as={BiSearch} h={5} w={5} />
```
Comment (m0nggh) [inline packages/frontend/src/components/DebouncedSearchInput/index.tsx:45]:
```suggestion
        <Icon as={BiSearch} boxSize={5} />
```

File: packages/frontend/src/components/DebouncedSearchInput/index.tsx [hunk 2]
```diff
@@ -13,31 +18,54 @@ export default function DebouncedSearchInput({
...
       </InputLeftElement>
       <Input
         type="text"
-        w="full"
+        w={{ base: '100%', md: '20rem' }}
+        maxW="100%"
```
Comment (m0nggh) [inline packages/frontend/src/components/DebouncedSearchInput/index.tsx]:
```suggestion
        w={{ base: '100%', md: '21rem' }}
        maxW="100%" // feels like this is not needed?
```
20rem looks like the icon is positioned too much to the right
image

File: packages/frontend/src/components/DebouncedSearchInput/index.tsx [hunk 3]
```diff
@@ -13,31 +18,54 @@ export default function DebouncedSearchInput({
...
+          as={BiSolidXCircle}
+          cursor="pointer"
+          opacity={0.6}
+          _hover={{ opacity: 1 }}
+          h={4}
+          w={4}
+          display={tempSearchValue ? 'block' : 'none'}
+          onClick={clearSearch}
+        />
+      </InputRightElement>
```
Comment (m0nggh) [inline packages/frontend/src/components/DebouncedSearchInput/index.tsx]:
```suggestion
      {tempSearchValue && (
        <InputRightElement h="100%">
          <Icon
            as={BiSolidXCircle}
            cursor="pointer"
            opacity={0.6}
            _hover={{ opacity: 1 }}
            onClick={clearSearch}
          />
        </InputRightElement>
      )}
```
cannot click on the right side when `tempSearchValue` is empty and `boxSize` is default 16px alr

File: packages/frontend/src/pages/Executions/components/SearchWithFilterInput.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    handleSearchInputChangeDebounced.cancel()
+  }, [onChange, handleSearchInputChangeDebounced])
+
+  // update padding of input element when filter element width changes.
+  useEffect(() => {
+    if (!filterRef.current) {
+      return
+    }
+    setInputPadding(filterRef.current.offsetWidth + 8)
+  }, [status])
```
Comment (m0nggh) [inline packages/frontend/src/pages/Executions/components/SearchWithFilterInput.tsx]: `[filterRef.current?.offsetWidth, status]`, seems like need to add the offsetWidth check or else when first typing, it doesn't update the padding properly hmm

File: packages/frontend/src/pages/Executions/components/SearchWithFilterInput.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+  return (
+    <InputGroup>
+      <InputLeftElement>
+        <Icon as={BiSearch} boxSize={5} />
+      </InputLeftElement>
+      <Input
+        textStyle="body-1"
+        w={{ base: '100%', md: '25rem' }}
+        maxW="100%"
```
Comment (m0nggh) [inline packages/frontend/src/pages/Executions/components/SearchWithFilterInput.tsx:67]: similar to above

File: packages/frontend/src/pages/Executions/components/SearchWithFilterInput.tsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+          as={BiSolidXCircle}
+          cursor="pointer"
+          opacity={0.6}
+          _hover={{ opacity: 1 }}
+          h={4}
+          w={4}
+          mr={2}
+          display={tempSearchValue ? 'block' : 'none'}
+          onClick={clearSearch}
+        />
```
Comment (m0nggh) [inline packages/frontend/src/pages/Executions/components/SearchWithFilterInput.tsx]: this one like cus got `mr` so like can still click when the `tempSearchValue` is empty but can remove the `h={4}` and `w={4}`

**Review by m0nggh — APPROVED** [review]
thks for the nice quick refactor, just need to make sure the close icon button looks ok based on the comments

**Comment by pregnantboy** [comment]
### Merge activity

* **Oct 17, 3:59 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/774).
* **Oct 17, 3:59 AM EDT**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/774).

**Pull request merged**